city_scrapers/spiders/il_finance_authority.py

```diff
@@ -0,0 +1,171 @@
+            dont_filter=True,
+            meta={"meeting_dict": meeting_dict},
+        )
+
+    def _parse_agenda_pdf(self, response):
+        try:
+            lp = LAParams(line_margin=0.1)
+            out_str = StringIO()
+            extract_text_to_fp(
+                inf=BytesIO(response.body),
+                outfp=out_str,
+                maxpages=1,
+                laparams=lp,
+                codec="utf-8",
+            )
+
+            pdf_content = out_str.getvalue().replace("\n", "")
+            # Remove duplicate spaces
+            clean_text = re.sub(r"\s+", " ", pdf_content)
+            # Remove underscores
+            clean_text = re.sub(r"_*", "", clean_text)
+            return clean_text
+
+        except PDFSyntaxError as e:
+            print("~~Error: " + str(e))
+
+    def _parse_meeting(self, response):
+        meeting_dict = response.meta["meeting_dict"]
+        title = meeting_dict["title"]
+        date = meeting_dict["date"]
+        time = meeting_dict["time"]
+        location = meeting_dict["location"]
+
+        meeting = Meeting(
+            title=title,
+            description="",
+            classification=self._parse_classification(title),
+            start=self._meeting_datetime(date, time),
+            end=None,
+            all_day=False,
+            time_notes="",
+            location=location,
+            links=self._parse_links(response.url, title),
+            source=self._parse_source(response),
+        )
+        meeting["status"] = self._get_status(meeting)
+        meeting["id"] = self._get_id(meeting)
+        yield meeting
+
+    def _meeting_datetime(self, date, time):
+        meeting_start = date + " " + time
+        meeting_start = meeting_start.replace(", ", ",").strip()
+        return datetime.strptime(meeting_start, "%b %d,%Y %I:%M %p")
+
+    def _get_pdf_link(self, item):
+        pdf_tag = item.css("td:nth-child(4) > a")
+        if not (pdf_tag):
+            return None
+        pdf_link = pdf_tag[0].attrib["href"]
+        return pdf_link
+
+    def _parse_title(self, item):
+        """Parse or generate meeting title."""
+        try:
+            title = item.css("td:nth-child(3)::text").extract_first()
+            return title
+        except TypeError:
+            return ""
+
+    def _parse_classification(self, title):
+        """Parse or generate classification from allowed options."""
+        if "Comm" in title:
+            return COMMITTEE
+        if "Board" in title:
+            return BOARD
+        return COMMISSION
+
+    def _parse_location(self, pdf_content):
+        try:
+            """Parse or generate location."""
+            address_match = re.search(
+                r"(?:in\s*the|at\s*the) .*(\. | \d{5})", pdf_content
+            )
+            address = address_match.group(0)
+            name = re.findall(r"(?:in\s*the|at\s*the).*?,", pdf_content)[0]
+        except Exception:
+            address = "Address Not Found"
```

Comment:
We can just return blank strings if that's the case, but it looks like the location is pretty consistent so it could be easier to use `_validate_location` and raise an exception if the location isn't a set address like we do here https://github.com/City-Bureau/city-scrapers/blob/c42620480f9184ca74885e2a3c6ef576faba3eb7/city_scrapers/spiders/chi_ssa_61.py#L65